### PR TITLE
More unit tests for scheduler

### PR DESCRIPTION
**Pull request by somtochiama**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
 /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Part of #80272 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by draveness — COMMENTED** [review]
/ok-to-test

**Comment by haosdent** [comment]
@SomtochiAma could you squash the commits ?

**Comment by SomtochiAma** [comment]
> @SomtochiAma could you squash the commits ?

Sure I will do that now.

File: pkg/scheduler/framework/v1alpha1/framework_test.go [hunk 1]
```diff
@@ -876,6 +876,454 @@ func TestFilterPlugins(t *testing.T) {
...
+				},
+				{
+					name: "TestPlugin 1",
+					inj:  injectedResult{PreBindStatus: int(Success)},
+				},
+			},
+			wantStatus: nil,
+		},
+		{
+			name: "ErrorErrorPreBindPlugin",
```
Comment (draveness) [inline pkg/scheduler/framework/v1alpha1/framework_test.go]: This test name looks weird, do we have a better name to describe this test?

**Review by draveness — COMMENTED** [review]
I think you could take the name from `TestFilterPlugins` here

File: pkg/scheduler/framework/v1alpha1/framework_test.go (hunk 1 above)
Comment (SomtochiAma) [reply pkg/scheduler/framework/v1alpha1/framework_test.go]: ErrorAndErrorPlugins - is this okay?

**Review by draveness — COMMENTED** [review]
Thanks.

You could squash the commits after Wei's approval

/lgtm
/assign huang-wei

**Comment by haosdent** [comment]
@SomtochiAma can squash commits.

File: pkg/scheduler/framework/v1alpha1/framework_test.go [hunk 2]
```diff
@@ -876,6 +876,454 @@ func TestFilterPlugins(t *testing.T) {
...
+				if err := registry.Register(pl.name, func(_ *runtime.Unknown, _ FrameworkHandle) (Plugin, error) {
+					return tmpPl, nil
+				}); err != nil {
+					t.Fatalf("Unable to register pre bind plugins: %s", pl.name)
+				}
+
+				configPlugins.PreBind.Enabled = append(
+					configPlugins.PreBind.Enabled,
+					config.Plugin{Name: pl.name})
+
```
Comment (Huang-Wei) [inline pkg/scheduler/framework/v1alpha1/framework_test.go]: Remove this empty line.

File: pkg/scheduler/framework/v1alpha1/framework_test.go [hunk 3]
```diff
@@ -876,6 +876,454 @@ func TestFilterPlugins(t *testing.T) {
...
+				tmpPl := pl
+				if err := registry.Register(pl.name, func(_ *runtime.Unknown, _ FrameworkHandle) (Plugin, error) {
+					return tmpPl, nil
+				}); err != nil {
+					t.Fatalf("Unable to register pre bind plugins: %s", pl.name)
+				}
+
+				configPlugins.PreBind.Enabled = append(
+					configPlugins.PreBind.Enabled,
+					config.Plugin{Name: pl.name})
```
Comment (Huang-Wei) [inline pkg/scheduler/framework/v1alpha1/framework_test.go]: Can you check this style of code in this file, and make it to:

```go
xyz = append(
    configPlugins.xyz,
    config.Plugin{},
)
```

File: pkg/scheduler/framework/v1alpha1/framework_test.go [hunk 4]
```diff
@@ -876,6 +876,454 @@ func TestFilterPlugins(t *testing.T) {
...
+				if err := registry.Register(pl.name, func(_ *runtime.Unknown, _ FrameworkHandle) (Plugin, error) {
+					return tmpPl, nil
+				}); err != nil {
+					t.Fatalf("Unable to register pre bind plugins: %s", pl.name)
+				}
+
+				configPlugins.Reserve.Enabled = append(
+					configPlugins.Reserve.Enabled,
+					config.Plugin{Name: pl.name})
+
```
Comment (Huang-Wei) [inline pkg/scheduler/framework/v1alpha1/framework_test.go]: Remove this empty line.

**Review by Huang-Wei — COMMENTED** [review]
Some style nits.

File: pkg/scheduler/framework/v1alpha1/framework_test.go (hunk 3 above)
Comment (SomtochiAma) [reply pkg/scheduler/framework/v1alpha1/framework_test.go]: I am not sure what you mean. Please, can you clarify further?

Comment (Huang-Wei) [reply pkg/scheduler/framework/v1alpha1/framework_test.go]: I mean the ending `)` is preferred to be put in a new line:

```go
xyz = append(
    configPlugins.xyz,
    config.Plugin{})
```

=>

```go
xyz = append(
    configPlugins.xyz,
    config.Plugin{},
)
```

**Pull request merged**